provider/oci/images.go

```diff
@@ -284,11 +300,56 @@ func NewInstanceImage(img ociCore.Image, compartmentID *string) (imgType Instanc
 
 	version, err := NewImageVersion(img)
 	if err != nil {
-		return imgType, err
+		return InstanceImage{}, "", err
 	}
 	imgType.Version = version
 
-	return imgType, nil
+	return imgType, arch, nil
+}
+
+// parseUbuntuImage returns the base and architecture of the returned image
+// from the OCI sdk.
+func parseUbuntuImage(img ociCore.Image) (corebase.Base, string, error) {
+	var (
+		arch string
+		base corebase.Base
+	)
+	// we need to split the provided OperatingSystemVersion, since
+	// it can contain information that does not belong to the base's
+	// channel.
+	// E.g: Canonical-Ubuntu-22.04-Minimal-aarch64-2023.08.27-0
+	// becomes:
+	//        OperatingSystem:        Canonical Ubuntu
+	//        OperatingSystemVersion: 22.04 Minimal aarch64
```

Comment:
This was copied from older code, and should not be valid with your fixes. Please validate.

Reply:
I rephrased this comment